units/stats_task_end.go

```diff
@@ -73,17 +75,27 @@ func (j *collectTaskEndDataJob) Run(ctx context.Context) {
 
 	var err error
 	if j.task == nil {
-		j.task, err = task.FindOneId(j.TaskID)
+		j.task, err = task.FindOneByIdAndExecution(j.TaskID, j.Execution)
 		j.AddError(err)
+		if err != nil {
```

Comment:
You should return regardless of if there are errors

Reply:
Fixed.